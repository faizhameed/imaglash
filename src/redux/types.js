const actionTypes = {
  UNSPLASH_INITIALISED: "UNSPLASH_INITIALISED",
  ON_PAGE_LOAD: "ON_PAGE_LOAD",
  SEARCH_QUERY_CHANGED: "SEARCH_QUERY_CHANGED",
  LOAD_MORE_DATA: "lOAD_MORE_DATA",
  RANDOM_IMAGE_TRIGGER: "RANDOM_IMAGE_TRIGGER",
  USER_CLICKS_IMAGE: "USER_CLICKS_IMAGE",
};

export default actionTypes;
