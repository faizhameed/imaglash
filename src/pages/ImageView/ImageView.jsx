import React from "react";
import "./ImageView.scss";
import { Link } from "react-router-dom";

const ImageView = () => {
  return (
    <div className="wrapper">
      <div className="img-view-container">
        <h2>Image Detail view</h2>
        <Link className="back-hm" to="/">
          X
        </Link>
      </div>
    </div>
  );
};

export default ImageView;
