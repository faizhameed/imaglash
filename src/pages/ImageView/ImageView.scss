.wrapper {
  background-color: rgba(0, 0, 0, 0.842);
  position: fixed;
  left: 0;
  top: 0;
  width: 100vw;
  height: 100vh;
  z-index: 1;
  .img-view-container {
    min-width: 300px;
    width: 50vw;
    height: 70vh;
    background-color: white;
    position: fixed;
    left: 50%;
    top: 50%;
    transform: translate(-50%, -50%);
    border-radius: 7px;
  }
  .back-hm {
    width: 30px;
    height: 30px;
    background-color: black;
    border-radius: 50%;
    text-decoration: none;
    position: absolute;
    right: -11px;
    top: -11px;
  }
}
