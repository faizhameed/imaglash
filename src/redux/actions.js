import actionTypes from "./types";
import { unsplash } from "../utils/common";
import { toJson } from "unsplash-js";

export const changeSearchQuery = (query) => {
  return function (dispatch) {
    dispatch({
      type: actionTypes.SEARCH_QUERY_CHANGED,
      payload: {
        results: [],
        query: "",
      },
    });
    if (unsplash) {
      unsplash.search
        .photos(query, 1, 9)
        .then(toJson)
        .then((json) => {
          dispatch({
            type: actionTypes.SEARCH_QUERY_CHANGED,
            payload: json,
            query: query,
          });
        });
    }
  };
};

export const updatePageNumber = (page, query) => {
  return function (dispatch) {
    if (unsplash) {
      unsplash.search
        .photos(query, page + 1, 9)
        .then(toJson)
        .then((json) => {
          dispatch({
            type: actionTypes.LOAD_MORE_DATA,
            payload: json,
            query,
          });
        });
    }
  };
};

export const setCollection = () => {
  return function (dispatch) {
    if (unsplash) {
      unsplash.search
        .photos("book", 1, 9)
        .then(toJson)
        .then((json) => {
          dispatch({
            type: actionTypes.ON_PAGE_LOAD,
            payload: json,
          });
        });
    }
  };
};

export const getRandomImage = () => {
  return function (dispatch) {
    unsplash.photos
      .getRandomPhoto({ query: "nature" })
      .then(toJson)
      .then((json) => {
        dispatch({
          type: actionTypes.RANDOM_IMAGE_TRIGGER,
          payload: json.urls.regular,
        });
      });
  };
};

export const userSelectImage = (data) => ({
  type: actionTypes.USER_CLICKS_IMAGE,
  payload: data,
});
