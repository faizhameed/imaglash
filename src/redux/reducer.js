import actionTypes from "./types";
import { combineReducers } from "redux";

const UNSPLASH_STATE = {
  unsplash: null,
  tags: [
    "Dog",
    "Cat",
    "Space",
    "Nature",
    "Business",
    "Office",
    "Coffee",
    "World",
    "Wildlife",
    "Beach",
    "Digital",
    "Meeting",
    "Cars",
    "Games",
    "Holiday",
  ],
  currentQuery: "Book",
  page: 1,
  collection: [],
  totalPages: 0,
  randomImageUrl: "",
  userImageSelection: null,
};

export function unsplashReducer(state = UNSPLASH_STATE, action) {
  switch (action.type) {
    case actionTypes.UNSPLASH_INITIALISED:
      return {
        ...state,
        unsplash: action.payload,
      };
    case actionTypes.ON_PAGE_LOAD:
      return {
        ...state,
        collection: [...action.payload.results],
        totalPages: action.payload.total_pages,
        page: 1,
      };

    case actionTypes.SEARCH_QUERY_CHANGED:
      return {
        ...state,

        collection: [...action.payload.results],
        page: 1,
        totalPages: action.payload.total_pages,
        currentQuery: action.query,
      };

    case actionTypes.LOAD_MORE_DATA:
      if (state.page < state.totalPages)
        return {
          ...state,
          page: state.page + 1,
          collection: [...state.collection, ...action.payload.results],
        };
      break;
    case actionTypes.RANDOM_IMAGE_TRIGGER:
      return {
        ...state,
        randomImageUrl: action.payload,
      };

    case actionTypes.USER_CLICKS_IMAGE:
      return {
        ...state,
        userImageSelection: action.payload,
      };
    default:
      return state;
  }
}

const rootReducer = combineReducers({
  unsplashReducer,
});

export default rootReducer;
