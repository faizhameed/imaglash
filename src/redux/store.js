import { createStore } from "redux";
import rootReducer from "./reducer";
import { applyMiddleware } from "redux";
import { createLogger } from "redux-logger";
import thunkMiddleware from "redux-thunk";

const logger = createLogger();
/**
 * we are creating global store and using the store in our root file to have
 * all components to access it
 *
 */

const middlewares = [thunkMiddleware];
/**
 * we need not run logger in production mode since it can show the state to users.
 * To improve security of out application we should only run logger in development mode
 */
if (process.env.NODE_ENV === "development") {
  middlewares.push(logger);
}
export const store = createStore(rootReducer, applyMiddleware(...middlewares));
