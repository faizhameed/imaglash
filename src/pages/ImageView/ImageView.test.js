import { shallow } from "enzyme";
import React from "react";
import ImageView from "./ImageView";
import { Provider } from "react-redux";
import { store } from "../../redux/store";

describe("Image View Page", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Provider store={store}>
        <ImageView />
      </Provider>
    );
  });
  it("expect to render Image View", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
