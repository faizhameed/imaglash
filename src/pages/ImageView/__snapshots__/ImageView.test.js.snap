// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image View Page expect to render Image View 1`] = `ShallowWrapper {}`;
